spec/osut_tests_spec.rb

```diff
@@ -972,19 +983,98 @@ module M
     expect(M.clean!).to eq(DBG)
+
+    # --- --- --- --- --- --- --- --- --- --- --- --- --- --- --- --- --- --- #
+    # Add electric heating to 1x slab.
+    entry = model.getSpaceByName("Entry way 1")
+    expect(entry.empty?).to be(false)
+    entry = entry.get
+    floor = entry.surfaces.select{ |s| s.surfaceType == "Floor" }
+    expect(floor.is_a?(Array)).to be(true)
+    expect(floor.size).to eq(1)
+    floor = floor.first
+
+    expect(entry.thermalZone.empty?).to be(false)
+    tzone = entry.thermalZone.get
+
+    # Retrieve construction.
+    expect(floor.isConstructionDefaulted).to be(false)
+    c = floor.construction
+    expect(c.empty?).to be(false)
+    c = c.get.to_LayeredConstruction
+    expect(c.empty?).to be(false)
+    c = c.get
+
+    # Recover single construction layer (concrete slab).
+    layers = OpenStudio::Model::MaterialVector.new
+    layers << c.layers.first
+    layers << c.layers.first
+    expect(c.layers.size).to eq(1)
+
+    # Generate construction with internal heat source.
+    cc = OpenStudio::Model::ConstructionWithInternalSource.new(model)
+    cc.setName("ihs")
+    expect(cc.setLayers(layers)).to be(true)
+    expect(cc.setSourcePresentAfterLayerNumber(1)).to be(true)
+    expect(cc.setTemperatureCalculationRequestedAfterLayerNumber(1)).to be(true)
+    expect(floor.setConstruction(cc)).to be(true)
+
+    availability = M.availabilitySchedule(model)
+    schedule = OpenStudio::Model::ScheduleConstant.new(model)
+    expect(schedule.setValue(22.78)).to be(true) # reuse cooling setpoint
+
+    # Create radiant electric heating.
+    ht = OpenStudio::Model::ZoneHVACLowTemperatureRadiantElectric.new(
+            model, availability, schedule)
+    ht.setName("radiant electric")
+    expect(ht.setRadiantSurfaceType("Floors")).to be(true)
+    expect(ht.addToThermalZone(tzone)).to be(true)
+    expect(tzone.setHeatingPriority(ht, 1)).to be(true)
+    found = false
+
+    tzone.equipment.each do |eq|
+      found = true if eq.nameString == "radiant electric"
+    end
+
+    expect(found).to be(true)
+
+    file = File.join(__dir__, "files/osms/out/seb_ihs.osm")
+    model.save(file, true)
+
+    # Regardless of the radiant electric heating installation, priority is
+    # given to the zone thermostat heating setpoint.
+    stpts = M.setpoints(entry)
+    expect(stpts[:heating]).to be_within(TOL).of(22.11)
+
+    # Yet if one were to remove the thermostat altogether ...
+    tzone.resetThermostatSetpointDualSetpoint
+    res = M.maxHeatScheduledSetpoint(tzone)
+    expect(res.is_a?(Hash)).to be(true)
+    expect(res.key?(:spt)).to be(true)
+    expect(res.key?(:dual)).to be(true)
+    expect(res[:spt ].nil?).to be(false)
+    expect(res[:spt]).to be_within(TOL).of(22.78) # radiant heating
+    expect(res[:dual]).to eq(false)
+
+    stpts = M.setpoints(entry)
+    expect(stpts[:heating].nil?).to be(false)
+    expect(stpts[:heating]).to be_within(TOL).of(22.78) # radiant heating
```

Comment:
Adding a **zone equipment** temperature setpoint test (independent of **zone thermostat** temperature setpoint).